lib/java/com/google/android/material/button/MaterialButton.java

```diff
@@ -663,9 +663,15 @@ private void updateIcon() {
       icon.setBounds(iconLeft, 0, iconLeft + width, height);
     }
 
-    if (iconGravity == ICON_GRAVITY_START || iconGravity == ICON_GRAVITY_TEXT_START) {
+    // Only update the icon position if it has changed
+    boolean iconStart = iconGravity == ICON_GRAVITY_START || iconGravity == ICON_GRAVITY_TEXT_START;
+    Drawable[] existingDrawables = TextViewCompat.getCompoundDrawablesRelative(this);
+    boolean hasDrawableStart = existingDrawables[0] != null;
+    boolean hasDrawableEnd = existingDrawables[2] != null;
+
+    if (iconStart && !hasDrawableStart) {
       TextViewCompat.setCompoundDrawablesRelative(this, icon, null, null, null);
```

Comment:
If icon is different from the existing start/end drawable, this update should happen

Reply:
@wcshi Woops, my bad, was too concerned with the positioning. Fixed.